python-checks/src/test/resources/checks/useOfAnyAsTypeHintImporting.py

```diff
@@ -0,0 +1,15 @@
+from typing import Any
+from abc import ABCMeta
+from useOfAnyAsTypeHintImported import ImportedParentWithMetaClass
+
+class LocalParent: ...
+
+class LocalParentWithMetaClass(metaclass=ABCMeta): ...
+
+class LocalWithMetaClassInherited(LocalParentWithMetaClass):
+    def local_inherited_foo(self) -> Any: # Noncompliant
+        ...
+
+class ImportedWithMetaClassInherited(ImportedParentWithMetaClass):
+    def imported_inherited_foo(self) -> Any: # FN SONARPY-2331
```

Comment:
For my understanding, is this because losing the metaclass information makes the `classSymbol.hasUnresolvedTypeHierarchy()` to be true? Or is it something else?

Reply:
for this specific case, it leads to the incorrect result of the `ClassSymbolImpl.hasSuperClassWithUnknownMetaClass` method